### PR TITLE
Cleanup table builder

### PR DESCRIPTION
**Pull request by ktf**

### CODE REVIEW

**Comment by ktf** [comment]
@aalkin could you have a second look at this? All the unit tests seem to pass and compilation time seems to be greatly improved.

File: Framework/Core/include/Framework/TableBuilder.h [hunk 1]
```diff
@@ -664,8 +671,8 @@ class TableBuilder
 
   void validate() const;
 
-  template <typename... ARGS>
-  auto makeBuilders(std::array<char const*, sizeof...(ARGS)> const& columnNames, size_t nRows)
+  template <typename... ARGS, size_t I = sizeof...(ARGS)>
```
Comment (aalkin) [inline Framework/Core/include/Framework/TableBuilder.h:674]: Does this improve compilation time somehow? Is it not equivalent to `sizeof...` in the array declaration?

**Comment by aalkin** [comment]
Other than the question above this seems very reasonable to me.

File: Framework/Core/include/Framework/TableBuilder.h (hunk 1 above)
Comment (ktf) [reply Framework/Core/include/Framework/TableBuilder.h:674]: What I see is that the sizeof...(ARGS) is left unexpanded (e.g. `sizeof...(int, float, ...)`), at least in some error messages, duplicating the already long list of template parameters. By having `size_t I` as a template parameter, it simply matches on the size of the array, when possible, resulting in a much smaller template name.

**Comment by ktf** [comment]
Checked on hyperloop as well. Merging.

**Pull request merged**